components/nexus-bootstrap/src/main/java/org/sonatype/nexus/bootstrap/jetty/ConnectorManager.java

```diff
@@ -0,0 +1,157 @@
+ */
+package org.sonatype.nexus.bootstrap.jetty;
+
+import java.util.IdentityHashMap;
+import java.util.Map;
+
+import org.eclipse.jetty.http.HttpScheme;
+import org.eclipse.jetty.server.ConnectionFactory;
+import org.eclipse.jetty.server.HttpConfiguration;
+import org.eclipse.jetty.server.HttpConnectionFactory;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.server.SslConnectionFactory;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+
+/**
+ * Jetty's internal connector manager to manage custom {@link ConnectorConfiguration}s.
+ */
+public class ConnectorManager
+{
+  private final Server server;
+
+  private final Map<String, String> properties;
+
+  private final Map<String, Object> namedBeans;
+
+  private final IdentityHashMap<ConnectorConfiguration, ServerConnector> managedConnectors;
+
+  public ConnectorManager(final Server server,
+                          final Map<String, String> properties,
+                          final Map<String, Object> namedBeans)
+  {
+    this.server = server;
+    this.properties = properties;
+    this.namedBeans = namedBeans;
+    this.managedConnectors = new IdentityHashMap<>();
+  }
+
+  public ServerConnector addConnector(final ConnectorConfiguration connectorConfiguration)
+  {
+    final HttpConfiguration httpConfiguration =
+        connectorConfiguration.customize(defaultHttpConfiguration(connectorConfiguration.getScheme()));
+
+    final ServerConnector serverConnector;
+    if (HttpScheme.HTTP == connectorConfiguration.getScheme()) {
+      final ConnectionFactory connectionFactory = new InstrumentedConnectionFactory(
+          new HttpConnectionFactory(httpConfiguration)
+      );
+      serverConnector = new ServerConnector(
+          server,
+          Integer.valueOf(property("http.acceptors", "-1")),
+          Integer.valueOf(property("http.selectors", "-1")),
+          connectionFactory
+      );
+    }
+    else if (HttpScheme.HTTPS == connectorConfiguration.getScheme()) {
+      verifyHttpsEnabled();
+      final SslContextFactory sslContextFactory = bean("sslContextFactory", SslContextFactory.class);
+      final ConnectionFactory connectionFactory = new InstrumentedConnectionFactory(
+          new SslConnectionFactory(sslContextFactory, "http/1.1")
+      );
+      serverConnector = new ServerConnector(
+          server,
+          Integer.valueOf(property("ssl.acceptors", "-1")),
+          Integer.valueOf(property("ssl.selectors", "-1")),
+          connectionFactory,
+          new HttpConnectionFactory(httpConfiguration)
+      );
+    }
+    else {
+      throw new IllegalArgumentException("Unsupported HTTP Scheme: " + connectorConfiguration.getScheme());
+    }
+    serverConnector.setHost(property("application-host", null));
+    serverConnector.setPort(connectorConfiguration.getPort());
+    serverConnector.setIdleTimeout(Long.valueOf(property("http.tineout", "30000")));
```

Comment:
Rather than duplicate the property names and defaults here, why not take the default HTTP connector (and HTTPS if it exists) and just copy over the relevant fields. ie. treat the default connector as a prototype which is used to populate any new connectors.